tests/test_env.py

```diff
@@ -237,7 +237,11 @@ def test_process_env_install_from_index_not_found(
     """
     Attempt to install a package that does not exist from a pip index. Assert the appropriate error is raised.
     """
-    with pytest.raises(env.PackageNotFound, match="Packages this-package-does-not-exist were not found in the given indexes."):
+    expected: str = (
+        "Packages this-package-does-not-exist were not found in the given indexes. "
+        "(Looking in indexes: %s)" % local_module_package_index
+    )
+    with pytest.raises(env.PackageNotFound, match=re.escape(expected)):
```

Comment:
The indexes are obtained by parsing the output of pip. This is a quite fragile method, so we need to test is properly. We should add tests for the cases where:
* There is more than one index configure
* Indexes are configured via environment variables (`PIP_INDEX_URL` and `PIP_EXTRA_INDEX_URL`).
* Also check if we don't miss a combination here.